### ISSUE TITLE
does nni support pruning my model and export onnx model

### ISSUE THREAD

**Issue by banbishan** [microsoft/nni#3089]
I am puzzled by this sentence in the nni document：
```
Since every module requires four functions for shape inference and module replacement, this is a large amount of work, we only implemented the ones that are required by the examples. If you want to speed up your own model which cannot supported by the current implementation, you are welcome to contribute.

For PyTorch we can only replace modules, if functions in forward should be replaced, our current implementation does not work. One workaround is make the function a PyTorch module.
```
can i pruning my own model and export onnx model, and then convert onnx model to tflite?

**Comment by QuanluZhang** [comment]
@banbishan thanks for asking. we will make it more clear in our doc.

For the first sentence, the current implementation of model speedup requires us to write shape inference logic, module replacement logic for each module. We only supported frequently used modules in CNN for now, and will increasingly support more.

Since we have module replacement logic when a module is compressed (e.g., with channels masked). It is not easy to replace functional op (e.g., F.relu) in `forward` function. so, we require users to use module (e.g., nn.Relu) instead of functional op if this op is compressed.

yes, you can prune your own model and [export onnx model](https://nni.readthedocs.io/en/stable/Compression/QuickStart.html#export-compression-result). This exported model is not speeded up, but the masked weights have been set to zero. If you want to make your pruned model faster, please apply [Model Speedup](https://nni.readthedocs.io/en/stable/Compression/ModelSpeedup.html). Then you can directly export the model after model speedup to ONNX using the practice in PyTorch

**Comment by banbishan** [comment]
Thank you for reply,  will do same practice on this.
